_source/_posts/2021-09-08-sonar-qube-dotnet.md

```diff
@@ -0,0 +1,146 @@
+- "Get up to speed on using #sonarqube and #dotnet in this post"
+- "Wondering how #sonarqube works in #dotnet? We've got you covered!"
+image: blog/sonar-qube-dotnet/sonar-qube-social.png
+type: awareness
+---
+
+When it comes to code quality and code security, [SonarQube](https://www.sonarqube.org/) is your teammate! This analysis tool is pretty straightforward to use, especially with some help from Docker.
+
+In this post, I'll show you how to run a Docker container with SonarQube to analyze the code of a simple ASP.NET Core 3.0 application. Code analysis is a critical component of app development because it can identify security issues and other tricky bugs that might be overlooked until unexpected behavior impacts end-users. Let's take a look. 
+
+## What Is SonarQube?
+
+In addition to evaluating the overall health of an application, SonarQube can highlight issues and bugs in the code. With a **Quality Gate** in place, you can _Clean As You Code_ and improve code quality systematically. Even more, this tool facilitates planning and allocating tasks for a group of programmers.
+
+SonarQube can help measure progress easily and validate the importance of technical work that's difficult to view or demonstrate, especially with non-technical customers. Let's explore some of the benefits of SonarQube and how you can implement it in your workflow.  
+
+**Table of Contents**{: .hide }
+* Table of Contents
+{:toc}
+
+## Key Features of SonarQube
+
+Before we get started on the tutorial, here are some of the capabilities of SonarQube that make it useful for automating code review in .NET Core: 
+- "Quality Gate" lets you set an acceptable quality standard for merging branches and releasing applications to production.
+- Code analysis is presented graphically and transparently. Programmers and non-programmers alike gain visibility into coding progress. 
+- You gain the insights needed to analyze bugs, security vulnerabilities, and code duplication. 
+- Quickly prioritize code changes and improvements.
+- SonarQube doesn't just pinpoint predefined code quality issues. If you wish, you can create custom rules to monitor specific issues. 
+- Support for more than 27 programming languages. SonarQube continues to offer a free and open source *Community Edition*, along with several *Editions* for commercial use. 
+
+## Prepare the SonarQube Environment 
+
+### Install Java and Docker
+
+To use SonarQube, first prepare your environment.  The requirements are pretty simple, a Java SDK for starters. We also recommend using Docker; a popular approach nowadays. If it's your first experience with Docker and you have doubts or questions,  there are many excellent online learning resources including [A Developer's Guide to Docker](/blog/2017/05/10/developers-guide-to-docker-part-1) on this blog.
+
+1. Since SonarScanner uses Java, you'll need a Java SDK (Java 11 or later) to `PATH`, and you'll add `JAVA_HOME` to your system variables.
```

Comment:
```suggestion
1. Since SonarScanner uses Java, you'll need to add a Java SDK (Java 11 or later) to `PATH`, and you'll add `JAVA_HOME` to your system variables.
```
missing verb